modules/rostests/apitests/shell32/CommandLineToArgvW.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+ * PROJECT:     ReactOS API tests
+ * LICENSE:     GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
+ * PURPOSE:     Test for CommandLineToArgvW
+ * COPYRIGHT:   Copyright 2023 Katayama Hirofumi MZ (katayama.hirofumi.mz@gmail.com)
+ */
+
+#include "shelltest.h"
+#include <atlstr.h>
+
+static VOID
+DoEntry(INT lineno, LPCWSTR cmdline, INT argc_minus_1, ...)
+{
+    va_list va;
+    va_start(va, argc_minus_1);
+
+    INT real_argc;
+    LPWSTR* real_argv = CommandLineToArgvW(cmdline, &real_argc);
+    INT real_argc_minus_1 = real_argc - 1;
+
+    INT m = min(argc_minus_1, real_argc_minus_1);
+    for (INT i = 0; i < m; ++i)
+    {
+        LPCWSTR arg1 = va_arg(va, LPCWSTR);
+        LPCWSTR arg2 = real_argv[i + 1];
+        CStringA str1 = wine_dbgstr_w(arg1);
+        CStringA str2 = wine_dbgstr_w(arg2);
+        ok(lstrcmpW(arg1, arg2) == 0, "Line %d: %s != %s\n", lineno, (LPCSTR)str1, (LPCSTR)str2);
```

Comment:
```suggestion
        ok(str1 == str2, "Line %d: %s != %s\n", lineno, (LPCSTR)str1, (LPCSTR)str2);
```

Why did you store it in CString's in the first place?

Reply:
`wine_dbgstr_w` looks to have a bug.